test/src/unit/mwc-textfield.test.ts

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {TextField} from '@material/mwc-textfield';
+import {cssClasses} from '@material/textfield/constants';
+import {html} from 'lit-html';
+
+import {fixture, TestFixture} from '../util/helpers'
+
+let element: TextField;
+let fixt: TestFixture;
+
+const basic = html`
+  <mwc-textfield></mwc-textfield>
+`;
+
+const required = html`
+  <mwc-textfield label="I am required" required></mwc-textfield>
+`;
+
+suite('mwc-textfield - basic:');
+
+beforeEach(async () => {
+  fixt = await fixture(basic);
+
+  element = fixt.root.querySelector('mwc-textfield')!;
+});
+
+test('initializes as an mwc-textfield', () => {
+  assert.instanceOf(element, TextField);
+});
+
+test('setting value sets on input', async () => {
+  element.value = 'my test value';
+
+  const inputElement = element.shadowRoot!.querySelector('input');
+  assert(inputElement, 'my test value');
+});
+
+afterEach(() => {
+  if (fixt) {
+    fixt.remove();
+  }
+});
+
+const isUiInvalid =
+    (element: TextField) => {
+      return !!element.shadowRoot!.querySelector(`.${cssClasses.INVALID}`);
+    }
+
+suite('mwc-textfield - validation:');
```

Comment:
Interesting, I haven't seen this flat style before. I've always seen tests nested within the suite callback, like it is in the mocha docs (https://mochajs.org/#tdd). I guess the advantage of doing it in this flat style is that there's less code indentation? But I think I find it harder to visually parse the code this way, because the suites boundaries aren't as obvious now.

Reply:
This is qunit style ([expand below](https://github.com/material-components/material-components-web-components/pull/414/files/207eb83fd7163c4434672d919e908b6aadd9313d#diff-a2a3b7b0c9c3b4b93b4aebf4e3ec3cfbR108)). It doesn't allow for nesting. I'd kinda prefer the nesting tbh, but all the tests were already written this way.

@azakus @sorvell @frankiefu: any reason as to why y'all chose `qunit` over `TDD`?

Reply:
Here are the mocha docs for qunit https://mochajs.org/#qunit

Reply:
affected parties are okay to switching to TDD